### PR TITLE
tickets/SP-1592

### PR DESCRIPTION
**Pull request by MelissaGraham**

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -6,7 +6,158 @@ Roadmap for Community Alert Filters with the ANTARES Broker
...
+
+Once a Rubin data release exists (DP2, DR1), the community filters will be able to query and use proprietary LSST data, such as multi-year light curves or the photo-z of potential host galaxies.
+
+Processing limits
+-----------------
+
+The computational resources will be shared across all community filters, and thus what is available for a given filter will depend on the number of filters, their throughput, and the actual number of alerts per visit.
+
+The LSST observing strategy will make one visit every ~30 seconds.
+Any filter which takes, on average during the night, longer than 30 seconds per visit (~10,000 alerts) to process alerts from one visit will fall behind.
```
Comment (ebellm) [inline index.rst]: ANTARES chains its filters, so the whole set of filters needs to take less than the visit + slew/readout time to stay ahead.

File: index.rst [hunk 2]
```diff
@@ -6,7 +6,158 @@ Roadmap for Community Alert Filters with the ANTARES Broker
...
+--------------
+
+See Section 3.5 of the Rubin Data Products Definitions Document (`LSE-163 <https://lse-163.lsst.io/>`_) for a list of alert packet contents, and `sdm-schemas.lsst.io/apdb.html <https://sdm-schemas.lsst.io/apdb.html>`_ has the as-implemented column names.
+
+The community filters will use only the contents of the LSST alert packet plus any additional or derived data added by ANTARES, such as cross-matches with external (non-Rubin) catalogs performed by ANTARES.
+
+The contents of the alert packets might evolve over Rubin Operations, thereby enabling new community filters.
+For example, the final set of time series features to be computed during Alert Production remains to be determined (`DMTN-118 <https://dmtn-118.lsst.io/>`_).
+
+Once a Rubin data release exists (DP2, DR1), the community filters will be able to query and use proprietary LSST data, such as multi-year light curves or the photo-z of potential host galaxies.
```
Comment (ebellm) [inline index.rst]: I'm not sure ANTARES has promised this capability at the filtering stage, particularly since it requires authenticating users to the Rubin identity system.

File: index.rst [hunk 3]
```diff
@@ -6,7 +6,158 @@ Roadmap for Community Alert Filters with the ANTARES Broker
...
+
+Follow-up feasibility
+---------------------
+
+The LSST Alerts: Key Numbers document (`DMTN-102 <https://dmtn-102.lsst.io/>`_) document predicts an average of 10,000 alert-triggering astrophysical sources per visit, with a maximum of >30,000 in Galactic fields.
+The community filters should significantly sub-select the number of alerts per visit.
+Filters that pass too many alerts do not sufficiently inform follow-up prioritization, which is the point of the filter.
+
+* Average number of passing alerts per visit: 20 alerts
+* Maximum number of passing alerts per visit: 100 alerts
```
Comment (ebellm) [inline index.rst:103]: While I agree with the motivation, are these constraints from ANTARES?

File: index.rst [hunk 4]
```diff
@@ -6,7 +6,158 @@ Roadmap for Community Alert Filters with the ANTARES Broker
...
+
+Machine-learned models
+----------------------
+
+Community filters can be based on a trained ML model.
+
+However, any community filters based on machine learning models would need to be trained and validated by volunteer members of the science community, training is beyond the scope of the Rubin CST.
```
Comment (ebellm) [inline index.rst]: maybe add a sentence about making sure models are technically feasible in ANTARES?  I don't know what packages you can import, for instance.

File: index.rst [hunk 5]
```diff
@@ -6,7 +6,158 @@ Roadmap for Community Alert Filters with the ANTARES Broker
...
-See the `Documenteer documentation <https://documenteer.lsst.io/technotes/index.html>`_ for tips on how to write and configure your new technote.
+Overview
+========
+
+Requirements on the Rubin system specify that users will have access to alerts via predefined filters (Appendix A), hereafter referred to as "community filters."
+These community filters are intended to provide a low barrier-to-entry starting point for community members wanting to start with alert science, and as such should cover a range of Rubin alert use cases.
+The Rubin Community Science team (CST) will take input from the community to define, implement, and maintain the community filters.
+The existence of community filters served by the CST do not preclude users from developing their own customized filters.
+The ANTARES alert broker provides the functionality for user-defined filters, and the ANTARES team has agreed to host these community filters, so that Rubin Operations can meet these requirements without constructing an Alert Filtering Service (`DMTN-226 <https://dmtn-226.lsst.io/>`_).
+
```
Comment (ebellm) [inline index.rst:18]: I'd add links to the ANTARES paper and to https://nsf-noirlab.gitlab.io/csdc/antares/antares/

**Review by ebellm — CHANGES_REQUESTED** [review]
several comments

File: index.rst [hunk 6]
```diff
@@ -6,7 +6,158 @@ Roadmap for Community Alert Filters with the ANTARES Broker
...
-================
 
-See the `Documenteer documentation <https://documenteer.lsst.io/technotes/index.html>`_ for tips on how to write and configure your new technote.
+Overview
+========
+
+Requirements on the Rubin system specify that users will have access to alerts via predefined filters (Appendix A), hereafter referred to as "community filters."
+These community filters are intended to provide a low barrier-to-entry starting point for community members wanting to start with alert science, and as such should cover a range of Rubin alert use cases.
+The Rubin Community Science team (CST) will take input from the community to define, implement, and maintain the community filters.
+The existence of community filters served by the CST do not preclude users from developing their own customized filters.
```
Comment (leannep) [inline index.rst]:
```suggestion
The existence of community filters served by the CST does not preclude users from developing their own customized filters.
```

File: index.rst [hunk 7]
```diff
@@ -6,7 +6,158 @@ Roadmap for Community Alert Filters with the ANTARES Broker
...
+The ANTARES alert broker provides the functionality for user-defined filters, and the ANTARES team has agreed to host these community filters, so that Rubin Operations can meet these requirements without constructing an Alert Filtering Service (`DMTN-226 <https://dmtn-226.lsst.io/>`_).
```
Comment (leannep) [inline index.rst]:
```suggestion
The ANTARES alert broker provides functionality for user-defined filters.  Rubin is partnering with the ANTARES broker to host its community filters so as to meet requirements without constructing a bespoke Alert Filtering Service (`DMTN-226 <https://dmtn-226.lsst.io/>`_).
```

File: index.rst [hunk 8]
```diff
@@ -6,7 +6,158 @@ Roadmap for Community Alert Filters with the ANTARES Broker
...
+
+#. Incorporate the Rubin science community’s input to develop, implement, test, and validate up to 20 Community Alert Filters in the ANTARES broker, starting with an initial set of ~5 filters during commissioning.
+#. Provide documentation and use-case examples for the community filters.
+#. Maintain and add to the set of community filters during Rubin Operations.
+
+These community filters will:
+
+* be designed to prevent redundancy among user-defined filters,
+* together as a set, serve a broad range of science interests,
+* individually, each meet a well-defined science goal,
```
Comment (leannep) [inline index.rst]: One or more well defined science goals? Some filters might serve > 1 science goal and still be well-defined

File: index.rst (hunk 2 above)
Comment (leannep) [reply index.rst]: I think Eric is correct, however, the SRD requirement text says: "Users may also query the LSST science database at any time for additional information that may be useful, such as the properties of static objects that are positionally close to the candidate transients." -- so we probably need to have this conversation with ANTARES

Comment (leannep) [reply index.rst]: Although all a Rubin user needs to do is set up their token on ANTARES to query the RSP -- the requirement does not state that the AFS must query the RSP to improve the result of the "community filter"

Comment (leannep) [inline index.rst]:
```suggestion
Once a Rubin data release exists (DP2, DR1), a Rubin data rights holder will be able to query and use proprietary LSST data, such as multi-year light curves or the photo-z of potential host galaxies to continue analysis of data returned by the community filters .
```

Comment (leannep) [reply index.rst]: .. added a suggestion based on this discussion

File: index.rst [hunk 9]
```diff
@@ -6,7 +6,158 @@ Roadmap for Community Alert Filters with the ANTARES Broker
...
+For example, the final set of time series features to be computed during Alert Production remains to be determined (`DMTN-118 <https://dmtn-118.lsst.io/>`_).
+
+Once a Rubin data release exists (DP2, DR1), the community filters will be able to query and use proprietary LSST data, such as multi-year light curves or the photo-z of potential host galaxies.
+
+Processing limits
+-----------------
+
+The computational resources will be shared across all community filters, and thus what is available for a given filter will depend on the number of filters, their throughput, and the actual number of alerts per visit.
+
+The LSST observing strategy will make one visit every ~30 seconds.
```
Comment (leannep) [inline index.rst]: An 30 sec image is made every 34 secs, including readout.

**Review by leannep — CHANGES_REQUESTED** [review]

File: index.rst (hunk 1 above)
Comment (ryanlauastro) [reply index.rst]: Thanks for the comment, Eric. In a discussion with Tom he mentioned that we would need to evaluate the global impact of a filter on the system. So the suggestion is to reword this to be more general about the time it takes to run all filters.

File: index.rst (hunk 3 above)
Comment (ryanlauastro) [reply index.rst:103]: The idea here is to provide a rough number for the goals of the community filters. Will add another sentence to clarify that (these are not constraints from ANTARES).

File: index.rst (hunk 2 above)
Comment (MelissaGraham) [reply index.rst]: Yes Ryan and I talked with Tom about this, and we're removing "Once a Rubin data release ... potential host galaxies." The current workflow is that ANTARES won't be querying out (takes too long) and that users do this (e.g., get more host info) with the filter results themselves via their authenticated access (e.g., can be programmatic via pyvo). But we don't need to say any of that here.

Comment (ryanlauastro) [reply index.rst]: From a discussion with Tom -- ANTARES shouldn't be running any queries since it slows things down significantly. However, users are able to query the databases after receiving their filtered alerts from ANTARES.

File: index.rst (hunk 4 above)
Comment (MelissaGraham) [reply index.rst]: Good point, we need to add a statement (or a subsection) about packages such as "The use of third-party software packages in community filters will be evaluated, and proposed filters that rely on potentially unstable or unmaintained packages will not be able to be implemented. " or similar

File: index.rst (hunk 3 above)
Comment (MelissaGraham) [reply index.rst:103]: They're not from ANTARES, but motivated by filter output usability -- a statement will be added.

File: index.rst (hunk 4 above)
Comment (ryanlauastro) [reply index.rst]: From a discussion with Tom, ANTARES can do ML models (but not active learning models), but that packages required for filters need to be vetted. Will add more detail on the package vetting.

Comment (MelissaGraham) [reply index.rst]: And yes, use of trained ML models is technically feasible for ANTARES (they're already doing it)

File: index.rst (hunk 8 above)
Comment (MelissaGraham) [reply index.rst]: Ah yeah, "each meet at least one well-defined science goal", is what we were going for -- will change

**Comment by ryanlauastro** [comment]
Add link to ANTARES catalogs in alert contents section: https://antares.noirlab.edu/catalogs

File: index.rst [hunk 10]
```diff
@@ -6,7 +6,158 @@ Roadmap for Community Alert Filters with the ANTARES Broker
...
 
-See the `Documenteer documentation <https://documenteer.lsst.io/technotes/index.html>`_ for tips on how to write and configure your new technote.
+Overview
+========
+
+Requirements on the Rubin system specify that users will have access to alerts via predefined filters (Appendix A), hereafter referred to as "community filters."
+These community filters are intended to provide a low barrier-to-entry starting point for community members wanting to start with alert science, and as such should cover a range of Rubin alert use cases.
+The Rubin Community Science team (CST) will take input from the community to define, implement, and maintain the community filters.
+The existence of community filters served by the CST does not preclude users from developing their own customized filters.
+The ANTARES alert broker provides functionality for user-defined filters.  Rubin is partnering with the ANTARES broker to host its community filters so as to meet requirements without constructing a bespoke Alert Filtering Service (`DMTN-226 <https://dmtn-226.lsst.io/>`_).
```
Comment (ryanlauastro) [inline index.rst]:
```suggestion
The (`ANTARES alert broker < https://nsf-noirlab.gitlab.io/csdc/antares/antares/ >`_) (`Matheson et al. (2021) < https://iopscience.iop.org/article/10.3847/1538-3881/abd703 >`_) provides functionality for user-defined filters.  Rubin is partnering with the ANTARES broker to host its community filters so as to meet requirements without constructing a bespoke Alert Filtering Service (`DMTN-226 <https://dmtn-226.lsst.io/>`_).
```

File: index.rst [hunk 11]
```diff
@@ -6,7 +6,158 @@ Roadmap for Community Alert Filters with the ANTARES Broker
...
+Overview
+========
+
+Requirements on the Rubin system specify that users will have access to alerts via predefined filters (Appendix A), hereafter referred to as "community filters."
+These community filters are intended to provide a low barrier-to-entry starting point for community members wanting to start with alert science, and as such should cover a range of Rubin alert use cases.
+The Rubin Community Science team (CST) will take input from the community to define, implement, and maintain the community filters.
+The existence of community filters served by the CST does not preclude users from developing their own customized filters.
+The (`ANTARES alert broker < https://nsf-noirlab.gitlab.io/csdc/antares/antares/ >`_) (`Matheson et al. (2021) < https://iopscience.iop.org/article/10.3847/1538-3881/abd703 >`_) provides functionality for user-defined filters.  Rubin is partnering with the ANTARES broker to host its community filters so as to meet requirements without constructing a bespoke Alert Filtering Service (`DMTN-226 <https://dmtn-226.lsst.io/>`_).
```
Comment (ryanlauastro) [inline index.rst]:
```suggestion
The (`ANTARES alert broker <https://nsf-noirlab.gitlab.io/csdc/antares/antares/>`_) (`Matheson et al. (2021) <https://iopscience.iop.org/article/10.3847/1538-3881/abd703>`_) provides functionality for user-defined filters.  Rubin is partnering with the ANTARES broker to host its community filters so as to meet requirements without constructing a bespoke Alert Filtering Service (`DMTN-226 <https://dmtn-226.lsst.io/>`_).
```

File: index.rst [hunk 12]
```diff
@@ -6,7 +6,158 @@ Roadmap for Community Alert Filters with the ANTARES Broker
...
+Overview
+========
+
+Requirements on the Rubin system specify that users will have access to alerts via predefined filters (Appendix A), hereafter referred to as "community filters."
+These community filters are intended to provide a low barrier-to-entry starting point for community members wanting to start with alert science, and as such should cover a range of Rubin alert use cases.
+The Rubin Community Science team (CST) will take input from the community to define, implement, and maintain the community filters.
+The existence of community filters served by the CST does not preclude users from developing their own customized filters.
+The (`ANTARES alert broker <https://nsf-noirlab.gitlab.io/csdc/antares/antares/>`_) (`Matheson et al. (2021) <https://iopscience.iop.org/article/10.3847/1538-3881/abd703>`_) provides functionality for user-defined filters.  Rubin is partnering with the ANTARES broker to host its community filters so as to meet requirements without constructing a bespoke Alert Filtering Service (`DMTN-226 <https://dmtn-226.lsst.io/>`_).
```
Comment (ryanlauastro) [inline index.rst]:
```suggestion
The `ANTARES alert broker <https://nsf-noirlab.gitlab.io/csdc/antares/antares/>`_ (`Matheson et al. 2021 <https://iopscience.iop.org/article/10.3847/1538-3881/abd703>`_) provides functionality for user-defined filters.  Rubin is partnering with the ANTARES broker to host its community filters so as to meet requirements without constructing a bespoke Alert Filtering Service (`DMTN-226 <https://dmtn-226.lsst.io/>`_).
```

File: index.rst (hunk 8 above)
Comment (ryanlauastro) [reply index.rst]:
```suggestion
* each meet at least one well-defined science goal,
```

File: index.rst (hunk 2 above)
Comment (ryanlauastro) [reply index.rst]:
```suggestion
```

File: index.rst (hunk 9 above)
Comment (ryanlauastro) [reply index.rst]:
```suggestion
The LSST observing strategy will make a 30 second image every 34 seconds, including readout.
```

File: index.rst (hunk 1 above)
Comment (ryanlauastro) [reply index.rst]:
```suggestion
Any filter which takes, on average during the night, longer than ~30 seconds per visit (~10,000 alerts) to process alerts from one visit may fall behind. The time it takes to run all filters will be evaluated to avoid significant delays. 
```

File: index.rst [hunk 13]
```diff
@@ -6,7 +6,157 @@ Roadmap for Community Alert Filters with the ANTARES Broker
...
+Any filter which takes, on average during the night, longer than 30 seconds per visit (~10,000 alerts) to process alerts from one visit will fall behind.
+
+* Maximum latency per visit: 30 seconds (average over the night)
+
+Follow-up feasibility
+---------------------
+
+The LSST Alerts: Key Numbers document (`DMTN-102 <https://dmtn-102.lsst.io/>`_) document predicts an average of 10,000 alert-triggering astrophysical sources per visit, with a maximum of >30,000 in Galactic fields.
+The community filters should significantly sub-select the number of alerts per visit.
+Filters that pass too many alerts do not sufficiently inform follow-up prioritization, which is the point of the filter.
```
Comment (ryanlauastro) [inline index.rst]:
```suggestion
Filters that pass too many alerts do not sufficiently inform follow-up prioritization, which is the point of the filter. The following provides suggested values for the number of passing alerts (note: these are not constraints from ANTARES):
```

File: index.rst [hunk 14]
```diff
@@ -6,7 +6,157 @@ Roadmap for Community Alert Filters with the ANTARES Broker
...
+
+* Average number of passing alerts per visit: 20 alerts
+* Maximum number of passing alerts per visit: 100 alerts
+
+Machine-learned models
```
Comment (ryanlauastro) [inline index.rst]:
```suggestion
Machine-learned models and Other Third-Party Software
```

File: index.rst (hunk 4 above)
Comment (ryanlauastro) [reply index.rst]:
```suggestion
The use of third-party software packages in community filters will be evaluated, and proposed filters that rely on potentially unstable or unmaintained packages will not be able to be implemented
```

File: index.rst [hunk 15]
```diff
@@ -6,7 +6,157 @@ Roadmap for Community Alert Filters with the ANTARES Broker
...
+----------------------
+
+Community filters can be based on a trained ML model.
```
Comment (ryanlauastro) [inline index.rst]:
```suggestion
Community filters can be based on a trained ML model. However, any community filters based on machine learning models would need to be trained and validated by volunteer members of the science community
```

File: index.rst (hunk 2 above)
Comment (MelissaGraham) [reply index.rst]: This entire sentence, "Once a Rubin data release exists ... returned by the community filters." was removed from the roadmap.
